rfcs/react-components/convergence/portal-compat.md

```diff
@@ -0,0 +1,166 @@
+# RFC: Portal compatibility
+
+[@layershifter](https://github.com/layershifter)
+
+## Summary
+
+This RFC proposes introduce compatibility between components in different Fluent UI versions that use [`ReactDOM.createPortal()`](https://reactjs.org/docs/portals.html) via:
+
+- usage of same values for `z-index`
+- usage of separate React context to assign CSS class that handles CSS variables from Fluent UI React v9
+
+## Problem statement
+
+Once components from Fluent UI React v9 (`@fluentui/react-components`) are put inside popovers & dialogs from Fluent UI React v8 (`@fluentui/react`) or Fluent UI React v0 (`@fluentui/react-northstar`) they don't have proper styles 💥
+
+Examples are below:
+
+- v8 & v9: https://codesandbox.io/s/v8-v9-no-compat-2nnt7
+- v0 & v9: https://codesandbox.io/s/v0-v9-no-compat-vw9e2
+
+Another problem is related to `z-index` values. Fluent UI React v8 (`@fluentui/react`) defines values that win over everything:
+
+```tsx
+export namespace ZIndexes {
+  export const Nav: number = 1;
+  /**
+   * @deprecated Do not use
+   */
+  export const ScrollablePane: number = 1;
+  export const FocusStyle: number = 1;
+  export const Coachmark: number = 1000;
+  export const Layer: number = 1000000;
+  export const KeytipLayer: number = 1000001;
+}
+```
+
+> Check [style-utilities/src/styles/zIndexes.ts](https://github.com/microsoft/fluentui/blob/36df8cdbdc8dd4b5989685f89b7d339ee5774bf7/packages/style-utilities/src/styles/zIndexes.ts#L1-L11)
+
+As v9 does not use such values components that use portals (`Popover`, `Menu`, `Tooltip`) appear **under** components from v8.
+
+Demo: https://codesandbox.io/s/v8layer-v9popover-q33b37
+
+## Detailed Design or Proposal
+
+### Use same `z-index` values
+
+- We can't change values in Fluent UI React v8 as this will be a breaking change
+- We want to make our components visible
+
+The simplest option that satisfies these requirements is to use same values in Fluent UI React v9.
+
+#### Pros and Cons
+
+- 👎 `z-index=1000000` is ugly
```

Comment:
another pro+con (combined) is that to solve the css vars, we need a context anyway, not storing an extra z-index in it doesn't change that fact